cmd/private_network_update.go

```diff
@@ -49,49 +48,102 @@ func (c *privateNetworkUpdateCmd) cmdPreRun(cmd *cobra.Command, args []string) e
 func (c *privateNetworkUpdateCmd) cmdRun(cmd *cobra.Command, _ []string) error {
 	var updated bool
 
-	ctx := exoapi.WithEndpoint(gContext, exoapi.NewReqEndpoint(account.CurrentAccount.Environment, c.Zone))
+	ctx := gContext
+	client, err := switchClientZoneV3(ctx, globalstate.EgoscaleV3Client, c.Zone)
+	if err != nil {
+		return err
+	}
 
-	privateNetwork, err := globalstate.EgoscaleClient.FindPrivateNetwork(ctx, c.Zone, c.PrivateNetwork)
+	resp, err := client.ListPrivateNetworks(ctx)
 	if err != nil {
-		if errors.Is(err, exoapi.ErrNotFound) {
-			return fmt.Errorf("resource not found in zone %q", c.Zone)
-		}
 		return err
 	}
 
+	pn, err := resp.FindPrivateNetwork(c.PrivateNetwork)
+	if err != nil {
+		return err
+	}
+
+	updateReq := v3.UpdatePrivateNetworkRequest{}
 	if cmd.Flags().Changed(mustCLICommandFlagName(c, &c.Description)) {
-		privateNetwork.Description = &c.Description
+		updateReq.Description = c.Description
 		updated = true
 	}
 
 	if cmd.Flags().Changed(mustCLICommandFlagName(c, &c.EndIP)) {
 		ip := net.ParseIP(c.EndIP)
-		privateNetwork.EndIP = &ip
+		updateReq.EndIP = ip
 		updated = true
 	}
 
 	if cmd.Flags().Changed(mustCLICommandFlagName(c, &c.Name)) {
-		privateNetwork.Name = &c.Name
+		updateReq.Name = c.Name
 		updated = true
 	}
 
 	if cmd.Flags().Changed(mustCLICommandFlagName(c, &c.Netmask)) {
 		ip := net.ParseIP(c.Netmask)
-		privateNetwork.Netmask = &ip
+		updateReq.Netmask = ip
 		updated = true
 	}
 
 	if cmd.Flags().Changed(mustCLICommandFlagName(c, &c.StartIP)) {
 		ip := net.ParseIP(c.StartIP)
-		privateNetwork.StartIP = &ip
+		updateReq.StartIP = ip
 		updated = true
 	}
 
-	if updated {
-		decorateAsyncOperation(fmt.Sprintf("Updating Private Network %q...", c.PrivateNetwork), func() {
-			if err = globalstate.EgoscaleClient.UpdatePrivateNetwork(ctx, c.Zone, privateNetwork); err != nil {
-				return
+	if cmd.Flags().Changed(mustCLICommandFlagName(c, &c.Option)) {
```

Comment:
this bit is basically the same as in `create` but not sure about the conventions u use over here...can i extract it somewhere to `private_network.go`?

Reply:
Yes, it would go in `private_network.go` file.
Make it a standalone function (not struct method) that accepts `[]string` and returns `*v3.PrivateNetworkOptions`.